test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs

```diff
@@ -185,6 +191,101 @@ protected NPath CompileAssembly (CompilerOptions options)
 
 		protected virtual NPath CompileCSharpAssemblyWithDefaultCompiler (CompilerOptions options)
 		{
+#if NETCOREAPP
+			return CompileCSharpAssemblyWithRoslyn (options);
+#else
+			return CompileCSharpAssemblyWithCodeDom (options);
+#endif
+		}
+
+#if NETCOREAPP
+		protected virtual NPath CompileCSharpAssemblyWithRoslyn (CompilerOptions options)
+		{
+			var parseOptions = new CSharpParseOptions (preprocessorSymbols: options.Defines);
+			var compilationOptions = new CSharpCompilationOptions (
+				outputKind: options.OutputPath.FileName.EndsWith (".exe") ? OutputKind.ConsoleApplication : OutputKind.DynamicallyLinkedLibrary
+			);
+			// Default debug info format for the current platform.
+			DebugInformationFormat debugType = RuntimeInformation.IsOSPlatform (OSPlatform.Windows) ? DebugInformationFormat.Pdb : DebugInformationFormat.PortablePdb;
```

Comment:
Is this correct? I thought core uses ppdb everywhere?

Reply:
I was matching the behavior of csc described at https://github.com/dotnet/roslyn/blob/master/docs/compilers/CSharp/CommandLine.md. However, you're right - it seems our SDK will default to passing `/debug:portable` to the compiler on all platforms. Maybe it would be better to match the SDK behavior here.

Reply:
Since we're translating csc commandline flags into roslin API options, I think it makes sense to match the csc commandline behavior, so tests that request `/debug:full` or `/debug:pdbonly` should get the platform-specific behavior. At the moment we don't have any tests that request debuginfo without specifying the format (viar `/debug+`). In that case the documented behavior is a bit ambiguous, but I believe csc still defaults to the platform-specific format.

Reply:
It might make sense to disable tests that rely on anything but a portable pdb though, since those are less relevant for .NET Core. Then again, it is possible to override the SDK behavior and ask for a specifig `DebugType`.

Reply:
Cecil supports only portable pdb cross platforms and this is not the right place to test pdb/ppdb differences as we don't care about such details in illinker. All tests should use only portable pdb for consistent behaviour.

Reply:
> All tests should use only portable pdb for consistent behaviour.

fyi We have tests for pdb, mdb, ppdb, and embedded pdb in order to ensure that the linker can handle all of them correctly.

Reply:
> We have tests for pdb, mdb, ppdb

right but these are tests which are more platform-specific not something we do for linker logic tests

Reply:
Whether we run tests that use different PDB formats is orthogonal to the default compiler behavior - in this change I'm just matching csc, and we could potentially disable any pdb tests that don't pass `"/debug:portable"` on core later if there's a reason not to run them. How does that sound?